alerting/src/main/kotlin/org/opensearch/alerting/WorkflowService.kt

```diff
@@ -0,0 +1,177 @@
+            .source(
+                SearchSourceBuilder()
+                    .query(bqb)
+                    .version(true)
+                    .seqNoAndPrimaryTerm(true)
+            )
+            .indices(chainedMonitor.dataSources.findingsIndex)
+        val searchResponse: SearchResponse = client.suspendUntil { client.search(searchRequest, it) }
+
+        // Get the findings docs
+        val findings = mutableListOf<Finding>()
+        for (hit in searchResponse.hits) {
+            val xcp = XContentFactory.xContent(XContentType.JSON)
+                .createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, hit.sourceAsString)
+            XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.nextToken(), xcp)
+            val finding = Finding.parse(xcp)
+            findings.add(finding)
+        }
+        // Based on the findings get the document ids
+        val indexToRelatedDocIdsMap = mutableMapOf<String, MutableList<String>>()
+        for (finding in findings) {
+            indexToRelatedDocIdsMap.getOrPut(finding.index) { mutableListOf() }.addAll(finding.relatedDocIds)
+        }
+        return indexToRelatedDocIdsMap
+    }
+
+    suspend fun searchMonitors(monitors: List<String>, size: Int, owner: String?): List<Monitor> {
+        val bqb = QueryBuilders.boolQuery().filter(QueryBuilders.termsQuery("_id", monitors))
+
+        val searchRequest = SearchRequest()
+            .source(
+                SearchSourceBuilder()
+                    .query(bqb)
+                    .version(true)
+                    .seqNoAndPrimaryTerm(true)
+                    .size(size)
+            )
+            .indices(ScheduledJob.SCHEDULED_JOBS_INDEX)
+
+        val searchResponse: SearchResponse = client.suspendUntil { client.search(searchRequest, it) }
+        return buildMonitors(searchResponse)
+    }
+
+    private fun buildMonitors(response: SearchResponse): List<Monitor> {
+        if (response.isTimedOut) {
+            log.error("Request for getting monitors timeout")
+            throw OpenSearchException("Cannot determine that the ${ScheduledJob.SCHEDULED_JOBS_INDEX} index is healthy")
+        }
+        val monitors = mutableListOf<Monitor>()
+        try {
+            for (hit in response.hits) {
+                XContentType.JSON.xContent().createParser(
+                    xContentRegistry,
+                    LoggingDeprecationHandler.INSTANCE, hit.sourceAsString
+                ).use { hitsParser ->
+                    val monitor = ScheduledJob.parse(hitsParser, hit.id, hit.version) as Monitor
+                    monitors.add(monitor)
+                }
+            }
+        } catch (e: Exception) {
+            log.error("Error parsing monitors: ${e.message}")
+            throw AlertingException.wrap(e)
+        }
+        return monitors
+    }
+
+    suspend fun getDocIdsPerFindingIndex(monitorId: String, workflowExecutionId: String): Map<String, List<String>> {
```

Comment:
javadocs

Reply:
This function will be removed since it's not used at all.